master/internal/usergroup/api_groups.go

```diff
@@ -0,0 +1,216 @@
+			Users:   model.Users(users).Proto(),
+		},
+	}, nil
+}
+
+func (a *ApiServer) GetGroups(ctx context.Context, req *apiv1.GroupSearchRequest,
+) (resp *apiv1.GroupSearchResponse, err error) {
+	// Detect whether we're returning special errors and convert to gRPC error
+	defer func() {
+		err = mapAndFilterErrors(err)
+	}()
+
+	groups, err := SearchGroups(ctx, req.Name, model.UserID(req.UserId), int(req.Offset),
+		int(req.Limit))
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiv1.GroupSearchResponse{
+		Groups: Groups(groups).Proto(),
+	}, nil
+}
+
+func (a *ApiServer) GetGroup(ctx context.Context, req *apiv1.GetGroupRequest,
+) (resp *apiv1.GetGroupResponse, err error) {
+	// Detect whether we're returning special errors and convert to gRPC error
+	defer func() {
+		err = mapAndFilterErrors(err)
+	}()
+
+	gid := int(req.GroupId)
+	g, err := GroupByID(ctx, gid)
+	if err != nil {
+		return nil, err
+	}
+
+	users, err := UsersInGroup(ctx, gid)
+	if err != nil {
+		return nil, err
+	}
+
+	gDetail := groupv1.GroupDetails{
+		GroupId: int32(g.ID),
+		Name:    g.Name,
+		Users:   model.Users(users).Proto(),
+	}
+
+	return &apiv1.GetGroupResponse{
+		Group: &gDetail,
+	}, nil
+}
+
+func (a *ApiServer) UpdateGroup(ctx context.Context, req *apiv1.UpdateGroupRequest,
+) (resp *apiv1.GroupWriteResponse, err error) {
+	// Detect whether we're returning special errors and convert to gRPC error
+	defer func() {
+		err = mapAndFilterErrors(err)
+	}()
+
+	oldGroup, err := GroupByID(ctx, int(req.GroupId))
+	if err != nil {
+		return nil, err
+	}
+
+	newName := oldGroup.Name
+	if req.Name != "" {
+		newName = req.Name
+	}
+	err = UpdateGroup(ctx, Group{
+		BaseModel: bun.BaseModel{},
+		ID:        int(req.GroupId),
+		Name:      newName,
+		OwnerID:   oldGroup.OwnerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(req.AddUsers) > 0 {
+		var users []model.UserID
+		for _, id := range req.AddUsers {
+			users = append(users, model.UserID(id))
+		}
+		err = AddUsersToGroup(ctx, int(req.GroupId), users...)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(req.RemoveUsers) > 0 {
+		var users []model.UserID
+		for _, id := range req.RemoveUsers {
+			users = append(users, model.UserID(id))
+		}
+		err = RemoveUsersFromGroup(ctx, int(req.GroupId), users...)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	users, err := UsersInGroup(ctx, int(req.GroupId))
+	if err != nil {
+		return nil, err
+	}
+
+	resp = &apiv1.GroupWriteResponse{
+		Group: &groupv1.GroupDetails{
+			GroupId: int32(oldGroup.ID),
+			Name:    newName,
+			Users:   model.Users(users).Proto(),
+		},
+	}
+
+	return resp, nil
+}
+
+func (a *ApiServer) DeleteGroup(ctx context.Context, req *apiv1.DeleteGroupRequest,
+) (resp *apiv1.DeleteGroupResponse, err error) {
+	// Detect whether we're returning special errors and convert to gRPC error
+	defer func() {
+		err = mapAndFilterErrors(err)
+	}()
+
+	err = DeleteGroup(ctx, int(req.GroupId))
+	if err != nil {
```

Comment:
nit: could do this ```if err := DeleteGroup(ctx, int(req.GroupId)); err != nil``` syntax and in some other places too

Reply:
I'm aware that's a commonly used syntax in Go, but I prefer not to use it because I don't like how much it buries the main action being done (e.g. `DeleteGroup`). I'm willing to change, though, if others have a strong preference here.